### PR TITLE
MQE-2316: MFTF SVC does not check newly added modules

### PR DESCRIPTION
**Pull request by soumyau**
# Description (*)

## Fixed Issues (if relevant)

1. magento/magento-semver#<issue_number>: Issue title

## Manual testing scenarios (*)

1. ...

2. ...

## Contribution checklist (*)

- [ ] Pull request has a meaningful description of its purpose
- [ ] All commits are accompanied by meaningful commit messages
- [ ] All new or changed code is covered with unit/integration tests (if applicable)
- [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/Analyzer/Mftf/ActionGroupAnalyzer.php [hunk 1]
```diff
@@ -189,6 +189,19 @@ public function analyze(Registry $registryBefore, Registry $registryAfter)
...
+        foreach ($newModuleEntities as $module => $entities) {
+            $this->findAddedEntitiesInModule(
+                $beforeEntities[$module] ?? [],
+                $entities,
+                self::MFTF_DATA_TYPE,
+                $this->getReport(),
+                ActionGroupAdded::class,
+                $module . '/ActionGroup'
+            );
+        }
```
Comment (jilu1) [inline src/Analyzer/Mftf/ActionGroupAnalyzer.php:204]: This check seems quite similar for all the analyzers. Can it be write as a generic function in `AbstractEntityAnalyzer.php`?

**Review by jilu1 — COMMENTED** [review]
Changes look good. Possibly a small refactory to avoid code duplication

File: src/Analyzer/Mftf/ActionGroupAnalyzer.php (hunk 1 above)
Comment (soumyau) [reply src/Analyzer/Mftf/ActionGroupAnalyzer.php:204]: findAddedEntitiesInModule is being reused from AbstractEntityAnalyzer, did you want another level of orchestration for it?

Comment (jilu1) [reply src/Analyzer/Mftf/ActionGroupAnalyzer.php:204]: I see. It's just a long params list that make the code look long. No need to change it to a function.

**Review by jilu1 — APPROVED** [review]

**Pull request merged**